File: Java/app/src/main/java/com/zivapp/notes/util/UtilIntent.java
package com.zivapp.notes.util;

import android.content.Context;
import android.content.Intent;

public class UtilIntent {

    public static void shareDataByIntent(Context context, String text) {
        Intent sendIntent = new Intent();
        sendIntent.setAction(Intent.ACTION_SEND);
        sendIntent.putExtra(Intent.EXTRA_TEXT, text);
        sendIntent.setType("text/plain");

        Intent shareIntent = Intent.createChooser(sendIntent, null);
        context.startActivity(shareIntent);
    }
}

File: Kotlin/app/src/main/java/com/zivapp/countandnote/util/UtilIntent.kt
package com.zivapp.countandnote.util

import android.content.Context
import android.content.Intent

class UtilIntent {
    companion object {
        fun shareDataByIntent(context: Context, text: String) {
            val sendIntent = Intent()
            sendIntent.action = Intent.ACTION_SEND
            sendIntent.putExtra(Intent.EXTRA_TEXT, text)
            sendIntent.type = "text/plain"
            val shareIntent = Intent.createChooser(sendIntent, null)
            context.startActivity(shareIntent)
        }
    }
}